src/c4/runtime_check.cc

```diff
@@ -0,0 +1,36 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   ds++/runtime_check.cc
+ * \author Kent Grimmett Budge
+ * \date   Wed Mar 28 07:58:48 2018
+ * \brief  Member definitions of class runtime_check
+ * \note   Copyright (C) 2018 Los Alamos National Security, LLC.
+ *         All rights reserved. */
+//---------------------------------------------------------------------------//
+
+#include <exception>
+
+#include "C4_Functions.hh"
+#include "ds++/DracoStrings.hh"
+
+namespace rtt_dsxx {
+
```

Comment:
Since this function doesn't wrap a MPI functionality, I think it should live in the `src/diagnostics` directory instead of `src/c4`.  
(The same can be said of other files/functions in c4, like ParallelUnitTest.  I'll add a personal task to review those files and move them.)

Reply:
I think that's a reasonable suggestion.